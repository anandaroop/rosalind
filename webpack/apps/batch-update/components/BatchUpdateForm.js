import React from 'react'
import styled from 'styled-components'
import { Link } from './Links'
import { LinkButton } from './Buttons'
import { colors } from './Layout'

class BatchUpdateForm extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      batchUpdateGenes: {}
    }
    this.handleCancelClick = this.handleCancelClick.bind(this)
  }

  handleCancelClick (e) {
    e.preventDefault()
    this.props.onCancel()
  }

  render () {
    const { selectedArtworkIds } = this.props
    const selectedArtworksCount = selectedArtworkIds.length

    return (
      <Wrapper>
        <Controls>
          <Link href='#' onClick={this.handleCancelClick} className='cancel'>Cancel</Link>
          <div>{selectedArtworksCount} works selected</div>
          <LinkButton onClick={e => console.log('TODO')}>Queue changes</LinkButton>
        </Controls>
      </Wrapper>
    )
  }
}

BatchUpdateForm.propTypes = {
  onCancel: React.PropTypes.func.isRequired,
  selectedArtworkIds: React.PropTypes.array.isRequired
}

const Wrapper = styled.div`
  padding: 18px 30px;
`

const Controls = styled.div`
  display: flex
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-bottom: 20px;
  border-bottom: solid 1px ${colors.grayLight}
`

export default BatchUpdateForm
