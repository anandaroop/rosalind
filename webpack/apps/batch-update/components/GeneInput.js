import React from 'react'
import styled from 'styled-components'
import { colors } from './Layout'

class GeneInput extends React.Component {
  render () {
    const { name, value } = this.props
    return (
      <Square value={value}>
        <Name value={value}>
          {name}
        </Name>
        <Value type='number' min='0' max='100' step='10' value={value === null ? '' : value} />
      </Square>
    )
  }
}

GeneInput.propTypes = {
  name: React.PropTypes.string.isRequired,
  value: React.PropTypes.number
}

const Square = styled.div`
  ${props => {
    if (parseInt(props.value) === 0) {
      return `
        border: solid 1px ${colors.red};
      `
    } else if (props.value) {
      return `
        border: solid 1px ${colors.purple};
      `
    } else {
      return `
        border: solid 1px ${colors.grayLighter};
      `
    }
  }}
  width: 140px;
  height: 140px;
  margin-right: 20px;
  margin-bottom: 20px;
  display: flex;
  flex-direction: column;
`
Square.displayName = 'Square'

const Name = styled.div`
  ${props => {
    if (parseInt(props.value) === 0) {
      return `
        background: ${colors.redLightest};
        border-bottom: solid 1px ${colors.red};
      `
    } else if (props.value) {
      return `
        background: ${colors.purpleLightest};
        border-bottom: solid 1px ${colors.purple};
      `
    } else {
      return `
        background: ${colors.grayLightest};
        border-bottom: solid 1px ${colors.grayLighter};
      `
    }
  }}
  flex-basis: 100px;
  padding: 0.5em 1em;
  overflow: hidden;
`
Name.displayName = 'Name'

const Value = styled.input`
  flex-basis: 40px;
  border-color: transparent;
  padding: 1em;
  &:focus {
    outline: solid 1px ${colors.gray};
  }
`
Value.displayName = 'Value'

export default GeneInput
