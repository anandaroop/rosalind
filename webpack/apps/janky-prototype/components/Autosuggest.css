.GeneAutosuggest,
.TagAutosuggest, 
.PartnerAutosuggest,
.FairAutosuggest {
  margin: 0;
}

.GeneAutosuggest ul.react-autosuggest__suggestions-list,
.TagAutosuggest ul.react-autosuggest__suggestions-list,
.PartnerAutosuggest ul.react-autosuggest__suggestions-list,
.FairAutosuggest ul.react-autosuggest__suggestions-list {
  list-style-type: none;
  margin: 1px 0 0 0;
  padding: 0;
  /*height: 300px;*/
  /*background: red;*/
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.GeneAutosuggest li.react-autosuggest__suggestion,
.TagAutosuggest li.react-autosuggest__suggestion,
.PartnerAutosuggest li.react-autosuggest__suggestion,
.FairAutosuggest li.react-autosuggest__suggestion {
  background: white;
  /*flex-basis: 50px;*/
  margin: 0 0 1px 0;
}

.GeneAutosuggest li.react-autosuggest__suggestion--focused,
.TagAutosuggest li.react-autosuggest__suggestion--focused,
.PartnerAutosuggest li.react-autosuggest__suggestion--focused,
.FairAutosuggest li.react-autosuggest__suggestion--focused {
  background: purple;
  color: white;
}

.GeneAutosuggest .GeneSuggestion,
.TagAutosuggest .TagSuggestion,
.PartnerAutosuggest .PartnerSuggestion,
.FairAutosuggest .FairSuggestion {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.GeneAutosuggest .GeneSuggestion,
.TagAutosuggest .TagSuggestion,
.PartnerAutosuggest .PartnerSuggestion,
.FairAutosuggest .FairSuggestion {
  min-height: 2em;
}

.GeneAutosuggest .GeneSuggestion-name,
.TagAutosuggest .TagSuggestion-name,
.PartnerAutosuggest .PartnerSuggestion-name,
.FairAutosuggest .FairSuggestion-name {
  flex: 1;
}

