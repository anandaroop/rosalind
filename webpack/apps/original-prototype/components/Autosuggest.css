.FairAutosuggest {
  margin: 0;
}

.FairAutosuggest ul.react-autosuggest__suggestions-list {
  list-style-type: none;
  margin: 1px 0 0 0;
  padding: 0;
  /*height: 300px;*/
  /*background: red;*/
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.FairAutosuggest li.react-autosuggest__suggestion {
  background: white;
  /*flex-basis: 50px;*/
  margin: 0 0 1px 0;
}

.FairAutosuggest li.react-autosuggest__suggestion--focused {
  background: purple;
  color: white;
}

.FairAutosuggest .FairSuggestion {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.FairAutosuggest .FairSuggestion {
  min-height: 2em;
}

.FairAutosuggest .FairSuggestion-name {
  flex: 1;
}
