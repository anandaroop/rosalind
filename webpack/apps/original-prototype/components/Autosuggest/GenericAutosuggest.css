ul.react-autosuggest__suggestions-list {
  /*list-style-type: none;*/
  /*margin: 1px 0 0 0;*/
  /*padding: 0;*/
  /*display: flex;*/
  /*flex-direction: column;*/
  /*justify-content: space-between;*/
}


li.react-autosuggest__suggestion {
  /*background: white;*/
  /*flex-basis: 50px;*/
  /*margin: 0 0 1px 0;*/
}

li.react-autosuggest__suggestion--focused {
  background: purple;
  color: white;
}
