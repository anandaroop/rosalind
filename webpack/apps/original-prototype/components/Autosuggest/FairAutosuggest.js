import React from 'react'
import GenericAutosuggest from './GenericAutosuggest'
import { matchFairs } from 'lib/rosalind-api'
import { getFairSuggestionValue, renderFairSuggestion } from './helpers'

function FairAutosuggest (props) {
  return (
    <GenericAutosuggest
      id='fair-autosuggest'
      placeholder={props.placeholder}
      fetchSuggestions={matchFairs}
      getSuggestionValue={getFairSuggestionValue}
      renderSuggestion={renderFairSuggestion}
      selectSuggestion={fair => { props.onSelectFair(fair) }}
      />
  )
}

FairAutosuggest.propTypes = {
  onSelectFair: React.PropTypes.func.isRequired
}

FairAutosuggest.defaultProps = {
  placeholder: 'Select a fair'
}

export { FairAutosuggest }
