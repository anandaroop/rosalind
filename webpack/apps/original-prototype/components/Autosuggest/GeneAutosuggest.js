import React from 'react'
import GenericAutosuggest from './GenericAutosuggest'
import { matchGenes } from 'lib/rosalind-api'
import { getGeneSuggestionValue, renderGeneSuggestion } from './helpers'

function GeneAutosuggest (props) {
  return (
    <GenericAutosuggest
      id='gene-autosuggest'
      placeholder={props.placeholder}
      fetchSuggestions={matchGenes}
      getSuggestionValue={getGeneSuggestionValue}
      renderSuggestion={renderGeneSuggestion}
      selectSuggestion={gene => { props.onSelectGene(gene) }}
      />
  )
}

GeneAutosuggest.propTypes = {
  onSelectGene: React.PropTypes.func.isRequired
}

GeneAutosuggest.defaultProps = {
  placeholder: 'Select a gene'
}

export { GeneAutosuggest }
