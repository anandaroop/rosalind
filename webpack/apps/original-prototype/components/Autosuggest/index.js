import React from 'react'
import GenericAutosuggest from './GenericAutosuggest'
import { matchGenes } from 'lib/rosalind-api'

function GeneAutosuggest2 (props) {
  return (
    <div className='GeneAutosuggest2'>
      <GenericAutosuggest
        id='gene'
        placeholder='Add a gene'
        fetchSuggestions={matchGenes}
        getSuggestionValue={gene => gene.name}
        renderSuggestion={gene => <div>{gene.name}</div>}
        selectSuggestion={gene => { props.onSelectGene(gene) }}
        />
    </div>
  )
}

export { GeneAutosuggest2 }
