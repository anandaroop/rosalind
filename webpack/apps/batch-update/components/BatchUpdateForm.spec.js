import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import BatchUpdateForm from './BatchUpdateForm'
import GeneInput from './GeneInput'
import { GeneAutosuggest } from './Autosuggest'
import ConfirmationModal from './ConfirmationModal'

let props, dismissHandler

beforeEach(() => {
  dismissHandler = jest.fn()
  props = {
    onCancel: dismissHandler,
    selectedArtworkIds: ['one', 'two', 'three']
  }
})

it('renders correctly', () => {
  const rendered = renderer.create(<BatchUpdateForm {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

describe('when the "Cancel" link is clicked', () => {
  it('calls the correct handler', () => {
    const wrapper = mount(<BatchUpdateForm {...props} />)
    const mockClickEvent = { preventDefault: jest.fn() }
    wrapper.find('a.cancel').simulate('click', mockClickEvent)
    expect(dismissHandler.mock.calls.length).toEqual(1)
  })

  it('resets the form state', () => {
    const wrapper = mount(<BatchUpdateForm {...props} />)
    wrapper.setState({
      geneValues: {
        'Kawaii': 70
      }
    })
    const mockClickEvent = { preventDefault: jest.fn() }
    wrapper.find('a.cancel').simulate('click', mockClickEvent)
    expect(wrapper.state('geneValues')).toEqual({})
  })
})

describe('when the "Queue" button is clicked', () => {
  it('it opens a confirmation modal', () => {
    const wrapper = mount(<BatchUpdateForm {...props} />)
    const mockClickEvent = { preventDefault: jest.fn() }
    wrapper.find('a.queue').simulate('click', mockClickEvent)
    expect(wrapper.find(ConfirmationModal).hasClass('modal-open')).toBe(true)
  })
})

it('adds genes', () => {
  const wrapper = mount(<BatchUpdateForm {...props} />)
  const component = wrapper.instance()
  component.onAddGene({ id: 'kawaii', name: 'Kawaii' })
  expect(component.state).toMatchObject({
    geneValues: {
      'Kawaii': null
    }
  })
})

describe('with currently added genes', () => {
  let wrapper, component

  beforeEach(() => {
    wrapper = mount(<BatchUpdateForm {...props} />)
    component = wrapper.instance()
    component.setState({
      geneValues: {
        'Kawaii': 0,
        'Animals': 100
      }
    })
  })

  it('renders the current genes', () => {
    expect(wrapper.find(GeneInput).length).toEqual(2)
    expect(wrapper.text()).toMatch('Kawaii')
    expect(wrapper.text()).toMatch('Animals')
  })

  it('updates gene values from integers', () => {
    component.onChangeGeneValue({ name: 'Kawaii', value: 70 })
    expect(component.state['geneValues']).toMatchObject({
      'Kawaii': 70
    })
  })

  it('updates gene values from number-ish strings', () => {
    component.onChangeGeneValue({ name: 'Kawaii', value: '70' })
    expect(component.state['geneValues']).toMatchObject({
      'Kawaii': 70
    })
  })

  it('nulls gene values from empty strings', () => {
    component.onChangeGeneValue({ name: 'Kawaii', value: '' })
    expect(component.state['geneValues']).toMatchObject({
      'Kawaii': null
    })
  })
})

it('includes a gene autocomplete widget', () => {
  const wrapper = mount(<BatchUpdateForm {...props} />)
  expect(wrapper.find(GeneAutosuggest).length).toEqual(1)
})
