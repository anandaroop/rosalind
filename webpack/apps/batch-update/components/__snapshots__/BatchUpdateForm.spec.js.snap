exports[`test renders correctly 1`] = `
<div
  className="igHpwR">
  <div
    className="YrORm">
    <a
      className="cancel gHceZM"
      href="#"
      onClick={[Function]}>
      Cancel
    </a>
    <div>
      3
       works selected
    </div>
    <a
      className="bMOQzN"
      onClick={[Function]}>
      Queue changes
    </a>
  </div>
  <div
    className="hfTCBQ" />
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-gene-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a gene"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-gene-autosuggest" />
    </div>
  </div>
</div>
`;
